x/oracle/types/currency_pair.go

```diff
@@ -31,48 +34,88 @@ func (cp *CurrencyPair) ValidateBasic() error {
 	if strings.ToUpper(cp.Quote) != cp.Quote {
 		return fmt.Errorf("incorrectly formatted quote string, expected: %s got: %s", strings.ToUpper(cp.Quote), cp.Quote)
 	}
+
+	if cp.Decimals <= 0 {
+		return fmt.Errorf("decimals must be greater than 0")
+	}
+
 	return nil
 }
 
-// String returns a string representation of the CurrencyPair, in the following form "ETH/BTC".
+// String returns a string representation of the CurrencyPair, in the following form "ETH/BTC/8" where 8 is the number of decimals.
 func (cp CurrencyPair) String() string {
+	return fmt.Sprintf("%s/%s/%d", cp.Base, cp.Quote, cp.Decimals)
+}
+
+// ID returns a string representation of the CurrencyPair, in the following form "ETH/BTC" where 8 is the number of decimals.
+// This function differs from String() which also includes the decimal information.
+func (cp *CurrencyPair) ID() string {
 	return fmt.Sprintf("%s/%s", cp.Base, cp.Quote)
 }
 
+// CurrencyPairStringToID takes a CurrencyPair string and returns the ID representation of it.
+func CurrencyPairStringToID(cpStr string) (string, error) {
+	cp, err := CurrencyPairFromString(cpStr)
+	if err != nil {
+		split := strings.Split(cpStr, "/")
+		if len(split) == 2 {
+			return cpStr, nil
+		}
+
+		return "", fmt.Errorf("invalid string provided: %w", err)
+	}
+
+	return cp.ID(), nil
+}
```

Comment:
IMO this should just return an error in the case when decimals is not included, feels like a footgun otherwise.